ms/MSOper/MSMetaData.cc

```diff
@@ -3516,15 +3524,35 @@ std::map<SubScanKey, MSMetaData::SubScanProperties> MSMetaData::_getSubScanPrope
     for ( ; miter!=mend; ++miter) {
         miter->second.meanExposureTime = Quantity(meanExposure[miter->first], eunit);
     }
-
-	uInt structSize = 3*sizeof(Double) + sizeof(Int);
-    uInt keySize = 4*sizeof(Int);
+    std::map<SubScanKey, map<uInt, vector<Double> > >::const_iterator iter = intervalSets.begin();
+    std::map<SubScanKey, map<uInt, vector<Double> > >::const_iterator end = intervalSets.end();
+    Unit unit = intervalTimes->getFullUnit();
+    for (; iter!=end; ++iter) {
+        SubScanKey ssKey = iter->first;
+        map<uInt, vector<Double> > spwToIntervalSets = iter->second;
+        map<uInt, vector<Double> >::const_iterator siter = spwToIntervalSets.begin();
+        map<uInt, vector<Double> >::const_iterator send = spwToIntervalSets.end();
+        for (; siter!=send; ++siter) {
+            uInt spw = siter->first;
+            vector<Double> myIntervals = siter->second;
```

Comment:
this and `spwToIntervalSets` above seem to be unnecessary copies, the data appears to only to be read so references should do